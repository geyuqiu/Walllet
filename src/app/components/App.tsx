import React from "react";
import {HashRouter} from "react-router-dom";

import {Routing} from "../router/Routing";
import {Footer} from "./Footer/Footer";

const notProd = process.env.NODE_ENV !== "production";

export const App = () => {
	const mainClassName = "bg-white min-h-full";
	const showDebugScreensOnDev: string = notProd
		? `debug-screens ${mainClassName}`
		: mainClassName;

	return (
		<HashRouter>
			{/*<Header />*/}
			<main className={showDebugScreensOnDev} role="main">
				<Routing />
			</main>
			<Footer />
		</HashRouter>
	);
};
