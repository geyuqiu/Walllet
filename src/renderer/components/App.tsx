import React from "react";
import { HashRouter } from "react-router-dom";

import { Routing } from "../router/Routing";
import { Footer } from "./Footer/Footer";
import { Header } from "./Header/Header";

const notProd = process.env.NODE_ENV !== "production";

export const App = () => {
	const showDebugScreensOnDev: string = notProd
		? "debug-screens bg-white min-h-70vh"
		: "";

	return (
		<HashRouter>
			<Header />
			<main className={showDebugScreensOnDev} role="main">
				<Routing />
			</main>
			<Footer />
		</HashRouter>
	);
};
