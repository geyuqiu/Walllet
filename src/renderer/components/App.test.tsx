import { render, screen } from "@testing-library/react";
import React from "react";

import { App } from "./App";

describe("App", () => {
	it("should render heading", () => {
		render(<App />);
		expect(screen.getByRole("heading")).toHaveTextContent("Hello");
	});

	it("should match snapshot", () => {
		const { container } = render(<App />);
		expect(container).toMatchSnapshot();
	});

	it("should show debug in TEST ENV", () => {
		render(<App />);
		expect(screen.getByRole("main")).toHaveClass("debug-screens");
	});
});
