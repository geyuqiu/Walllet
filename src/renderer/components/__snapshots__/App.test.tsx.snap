// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div>
  <header
    class="bg-green-700 p-6 min-h-20vh flex items-center justify-between flex-wrap"
  >
    <nav
      class="block"
    >
      <a
        aria-current="page"
        class="active"
        href="#/"
      >
        <span
          class="inline-block text-gray-800 hover:text-gray-600 mr-4"
          data-testid="link__wallet"
        >
          Wallet
        </span>
      </a>
      <a
        href="#/fee"
      >
        <span
          class="inline-block text-gray-800 hover:text-gray-600 mr-4"
          data-testid="link__fee"
        >
          Fee
        </span>
      </a>
    </nav>
  </header>
  <main
    class="debug-screens bg-white min-h-70vh"
    role="main"
  >
    <div>
      []
    </div>
  </main>
  <footer
    class="bg-gray-100 min-h-10vh flex items-center justify-center"
  >
    <span>
      2020 Â© ARK.io | All rights reserved
    </span>
  </footer>
</div>
`;
